### ISSUE TITLE
bug(Storybook): fix stories which error on render

### ISSUE THREAD

**Issue by uhLeeshUh** [qri-io/desktop#564]
**Overview of the issue:**
Currently the following stories error when running Storybook:

1. `LineDiff: 'LineDiff: +1-3'`: there is a TypeError for an `inserts` property that is undefined at runtime. Will likely involve stronger typing and handling the undefined case in code
<img width="1434" alt="Screen Shot 2020-07-10 at 1 28 26 PM" src="https://user-images.githubusercontent.com/31083146/87191316-694ee700-c2b1-11ea-974c-5393f8a35a17.png">

2. `Table: 'basic mixed edits', 'column swap', 'column swap & edits'`: all 3 stories have the same issue with `reduce` not being a property of `undefined`. Likely will involve handling case where passed array is `undefined`
<img width="1440" alt="Screen Shot 2020-07-10 at 1 32 03 PM" src="https://user-images.githubusercontent.com/31083146/87191571-ee3a0080-c2b1-11ea-813a-22cb17c68fed.png">

3. `Dataset: 'details subtext'`: no apparent console errors for this one, but it just renders a blank page
<img width="1437" alt="Screen Shot 2020-07-10 at 1 40 05 PM" src="https://user-images.githubusercontent.com/31083146/87192072-e595fa00-c2b2-11ea-9836-d8fddd1f8bd0.png">

4. `Chrome: Navbar: 'Standard'`: `store` doesn't appear to be in context for the `NavBar` component. Likely need to examine redux store config for this component.
<img width="1434" alt="Screen Shot 2020-07-10 at 1 41 37 PM" src="https://user-images.githubusercontent.com/31083146/87192707-470a9880-c2b4-11ea-8b45-86813b01dc16.png">

5. `Modals: 'Search'`: an error exists for using `withRouter` outside of a `<Router>` component
<img width="1437" alt="Screen Shot 2020-07-10 at 2 00 45 PM" src="https://user-images.githubusercontent.com/31083146/87193447-f5630d80-c2b5-11ea-830f-7381574c5468.png">

6. `Network: 'Home'`: an error exists for using `withRouter` outside of a `<Router>` component
<img width="1431" alt="Screen Shot 2020-07-10 at 2 00 54 PM" src="https://user-images.githubusercontent.com/31083146/87193459-ff850c00-c2b5-11ea-8b8a-824238da9de9.png">

**Comment by uhLeeshUh** [comment]
👋 @ramfox if I understood what you mentioned in standup correctly, it sounded like you think most of the issues we're seeing above are because we're using the connected versions of components (called with HOCs) instead of the base components, yeah? That would make sense to me for why we're seeing issues using `withRouter` or `Connect`. If so, I'm happy to play around with that idea and see if we can get a quick win. If it ends up being more problematic than that, I can post an update here and then maybe pull in this issue into a later sprint?

**Comment by ramfox** [comment]
Great! This is so helpful.

So I believe most of these issues stem from a large refactoring we did! If you take a look at our folder structure, even though we are a redux-react app, we have no "containers" folder. That's because we refactored our app to more containers into the components file page itself, and have the default export be the "containerized" version of the component (connected to redux, so the store, reducers, and actions).

For example, the NavBar component:

https://github.com/qri-io/desktop/blob/a1775125067bbf764808b66cc67817ced27d1f4c/app/components/nav/Navbar.tsx#L49-L72

There is a `NavBarComponent`, which is where the basic component is defined. It assumes that when it is called it will have all the contents of the `props`, but doesn't care about where they get it from.

But the `default export` is the output of the `connectComponentToPropsWithRouter ` function. This is a custom component (if you want to examine further/read about what the expected params are, take a look [here](https://github.com/qri-io/desktop/blob/master/app/utils/connectComponentToProps.ts)) that connects the component to redux, and returns a `Container`.

For Storybook, we don't have a state tree or a reducer etc, so we need to make sure we are using the `component` version of the component, not the default `containerized` version, when we import the components. The fix here would be to change those imports in storybook, and make sure we are passing in whatever props are expected, properly. EG, `NavBar` expects route props, so we have to figure out how to get em and give em:

```
import React from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import { NavbarComponent } from '../app/components/nav/Navbar'

export default {
  title: 'Chrome | Navbar',
  parameters: {
    notes: ''
  }
}

export const std = () => {
  return (
    <div style={{ width: 800, margin: '10px auto' }}>
      <Router>
        <Route render={(props) => {
          return <NavbarComponent setModal={() => console.log("set modal!")} {...props} />
        }}/>
      </Router>
    </div>
  )
}

std.story = {
  name: 'Standard',
  parameters: { note: 'basic navbar with no props' }
}

```
